harness/determined/cli/agent.py

```diff
@@ -76,6 +76,18 @@ def list_slots(args: argparse.Namespace) -> None:
         if r["container_id"]
     }
 
+    def get_task_name(containers: Dict[str, Any], slot: Dict[str, Any]) -> str:
+        if slot["container"] and slot["container"]["id"] in containers:
+            return str(containers[slot["container"]["id"]]["name"])
+        if slot["container"] and (
+            "determined-master-deployment" in slot["container"]["id"]
+            or "determined-db-deployment" in slot["container"]["id"]
+        ):
+            return "Determined System Task"
+        if slot["container"]:
+            return "Non-Determined Task"
```

Comment:
since we already know this information, would it be nicer for the user if we actually print the task name, prefixed with these, e.g.:

`Determined system task: determined-master-deployment`
and `Non-determined task: user-container-name`

then they'll immediately know *what* is taking the slot.

Reply:
Currently have the name of the pod as `allocation-id` but I'll change it to read this and for `allocation-id` to read "OCCUPIED" for non-det tasks.